pyflagser/_utils.py

```diff
@@ -0,0 +1,72 @@
+"""Utility functions for flag matrices"""
+
+import numpy as np
+
+
+def _extract_unweighted_graph(adjacency_matrix):
+    # Extract vertices and give them weight one
+    vertices = np.ones(adjacency_matrix.shape[0], dtype=np.float)
+
+    # Extract edges indices
+    if isinstance(adjacency_matrix, np.ndarray):
+        # Off-diagonal mask
+        mask = np.logical_not(np.eye(adjacency_matrix.shape[0], dtype=bool))
+
+        # Data mask
+        mask = np.logical_and(adjacency_matrix, mask)
+
+        edges = np.argwhere(mask)
+    else:
```

Comment:
I think something morally identical to the `argwhere` solution works for sparse matrices too. I've just tried it for CSR, CSC, COO, and LIL and observed between 2x and 5x speedup in all formats but COO by running instead:
```
mask = np.stack(np.nonzero(adjacency_matrix)).T
edges = mask[mask[:, 0] != mask[:, 1]]  # Removes diagonal elements a posteriori
```
With COO, I get basically the same performance. I suspect the problem is the conversion to COO.

Reply:
To reproduce:
```
N = 20
adjacency_matrix = np.random.random((N, N))  #random adjacency matrix of floats
rows = np.random.choice(range(N), int(N**2/2))
cols = np.random.choice(range(N), int(N**2/2))
adjacency_matrix[rows, cols] = 0  # Set ~1/2 of all entries to 0
```
and then convert to various sparse formats.

Reply:
Note: `argwhere` actually works here too it seems, but is fractionally slower for some reason.

Reply:
Thanks!